Automattic/jetpack
# Protect card: Add the Scan/Threats status column (with tooltip)

This PR populates the Scan/Threats section on the My Jetpack Protect card with the current scan status from Protect.




See Design P2: p1HpG7-rFA-p2 (Figma link is within the post)

Screenshots are shown below in the testing instructions.

## Proposed changes:

* Move the info tooltip into it's own component.
* Create a component for the Scan/Threats section, rendering the current Protect scan status, per Figma design

**Note:** Adding a "loading/currently scanning" state in the My Jetpack Protect product card is not included in this PR and will be handled in a future PR.  We need to add the Scan API js-package first.

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-aP6-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






You can reference the link to the Figma design in the design P2 post: p1HpG7-rFA-p
(tampermonkey does not shorthand the direct Figma link for some reason, so you'll need to get it from the design post).

- Checkout this branch via the Jetpack Beta plugin or your local dev environment (In this case I would suggest Jetpack Beta so you can more easily start with a fresh site)
    - In the Jetpack Beta plugin, run this branch on the Jetpack plugin **and** the Protect Plugin
- Open your JN site to get the SSH credentials, then SSH into the JN site.
- OR: Edit the plugin file in the editor by going to: `/wp-admin/plugin-editor.php`
- Edit the Akismet plugin file to trigger a threat:
    - Edit the `akismet.php` file: ` vim htdocs/wp-content-plugins/akismet/akismet.php`
    - In the top comment section, change the "`Version:`" to: `3.1.4`
    - Also on line 40, change it to: `define( 'AKISMET_VERSION', '3.1.4' );`
- Open the My Jetpack page.
- Look at the Protect product card. The "Scan" status column should show "Off", since the site is not connected: (See screenshot):
![Screen Shot on 2024-07-05 at 11-22-27](https://github.com/Automattic/jetpack/assets/11078128/94e57120-2be4-432b-950d-72edbcbb49e7)
- Click "Protect" in the Jetpack menu to go to the Protect page.  Select the Free plan and wait for the scan to complete.
**Note:** Adding a "loading/currently scanning" state in the My Jetpack Protect product card is not included in this PR and will be handled in a future PR.
- Once the scan has completed, go to My Jetpack (or refresh the My Jetpack page). 
- You should see 1 Threat in the Protect product card and it should match the Figma design.
![Screen Shot on 2024-07-10 at 23-36-40](https://github.com/Automattic/jetpack/assets/11078128/6bca4713-1ed9-4ed2-baa4-befd88080cea)
- Now edit the Akismet plugin file again (`akismet.php`):
- Add the following line anywhere inside the file, and then save:
        - `$eicar = "X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-CRITICAL-ANTIVIRUS-TEST-FILE!$H+H*";`
- Go to the Protect page.  Click "Upgrade Jetpack Protect now" and purchase with credits.
- After upgrade purchased, Protect should be scanning the site again.  Wait for the scan to complete.
Again note: adding a "loading/currently scanning" state in the My Jetpack Protect product card is not included in this PR and will be handled in a future PR.
- Once scan is completed, go to My Jetpack and view the Protect card.  It should show 1 threat (1 Critical threat), with a tooltip. (See screenshot):
![Screen Shot on 2024-07-04 at 15-03-56](https://github.com/Automattic/jetpack/assets/11078128/419ae970-60cc-4484-98cf-98f1a7df8561)
- Go to the protect page, and click to "Auto fix" the threat.  Wait for the threat fix to complete successfully.
- Once the threat is fixed, go to the My Jetpack page. View the protect card. It should show "Secure". (See screenshot):
![Screen Shot on 2024-07-05 at 12-37-54](https://github.com/Automattic/jetpack/assets/11078128/80c5a35a-f045-4a19-9a02-6183634f254b)
- Go to wordpress.com Store Admin and remove the Scan subscription you purchased for the site.
- Go back to the My Jetpack page and view the Protect card. It should say "Partial" with tooltip (See screenshot):
![Screen Shot on 2024-07-05 at 12-45-42](https://github.com/Automattic/jetpack/assets/11078128/7b0a390c-a0d5-4371-96d8-638529b9700b)
- Go to the Plugins page, and deactivate Protect.
- Go back to My Jetpack and the Protect card should Show "Off" (See screenshot):
![Screen Shot on 2024-07-05 at 12-50-39](https://github.com/Automattic/jetpack/assets/11078128/218ca06c-c8ed-43a1-bb60-3a730790f6f7)
